### DISCUSSION TITLE
Unable to create AppxPackage with all the required dependencies

### DISCUSSION THREAD

**Discussion by harvinders** [microsoft/WindowsAppSDK#2317] [Q&A] [answered]
I recently converted my WinAppSDK packaged solution with waproj to single csproj. I am now unable to create the correct AppxPackage.

I followed basic instructions at https://docs.microsoft.com/en-us/windows/apps/package-and-deploy/ci-for-winui3?pivots=winui3-packaged-csharp#building-from-command-line

The Dependencies folder only has `Microsoft.WindowsAppRuntime.1.0.msix` and missing `Microsoft.VCLibs.x64.14.00.appx` and `Microsoft.VCLibs.x64.14.00.Desktop.appx`.

I am using VisualStudio 17.1.1, is it a known issue or am I missing something?

Running the app results in following errors in eventviewer
```
Application: AppName.exe
CoreCLR Version: 6.0.322.12309
.NET Version: 6.0.3
Description: The process was terminated due to an unhandled exception.
Exception Info: System.DllNotFoundException: Unable to load DLL 'Microsoft.ui.xaml.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
   at Raya.Program.Main(String[] args)

Faulting application name: Raya.exe, version: 1.0.0.0, time stamp: 0x6216d2c0
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1387, time stamp: 0x0b9a844a
Exception code: 0xe0434352
Fault offset: 0x0000000000034f69
Faulting process id: 0x1bf0
Faulting application start time: 0x01d8404815997dc3
Faulting application path: C:\Program Files\WindowsApps\9AF2D1F1-CE1A-4B25-B0E8-DC28FD167C98_1.0.22084.140_x64__145zpcdw5xgtc\Raya.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 301f3899-b66b-44af-99d0-d98761e26b5e
Faulting package full name: 9AF2D1F1-CE1A-4B25-B0E8-DC28FD167C98_1.0.22084.140_x64__145zpcdw5xgtc
Faulting package-relative application ID: App

``

**Comment by Scottj1s** [comment, marked as answer]
@harvinders  Through v1.0.1, internal Windows App SDK binaries require the CRT to be deployed (either via VCLibs.Desktop framework package for packaged apps, or VCRedist*.exe for unpackaged apps).  A C# single-project app like yours can turn that behavior on by adding this property to the csproj:
**`<WinUISDKReferences>true</WinUISDKReferences>
`**
With v1.0.2, Windows App SDK binaries will no longer dynamically link to CRT libraries, so this won't be a problem and you can remove the property at that point.

See also #2117

**Reply by harvinders** [reply]
Compared to packaging using the two-project solution, for the single-project solution, I noticed that the size of the package has grown from 120MB to 188 MB for my app. It seems lot of .net assemblies are getting added to the msix package now (irrespective of `<WinUISDKReferences>true</WinUISDKReferences>`). Is this intended @Scottj1s ?
